From 1bd4ba4926d95ad6364821f994856b9a71a7214d Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 04 Jul 2024 09:44:48 +0300
Subject: [PATCH] [SW-186071] gaudi3: print wdt reset reason

The wdt reset reason is stored into cold reset
flops by arcmgmt and this information is printed
in ppboot which helps in identifiying the arc
responsible for wdt reset.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I8da347d7f35d0864325b7e82bbfdc0c19af2c65e
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 200e134..d6a1909 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -336,3 +336,24 @@
 {
 	return true;
 }
+
+void wd_cause_dump(void)
+{
+	struct gaudi3_cold_rst_data cold_rst_data;
+
+	cold_rst_data.data = regmap_readl(mmCOLD_RST_DATA);
+
+	if (cold_rst_data.wd_rst_cause_arcevent)
+		preboot_puts("wd cause: ARCEVENT heartbeat\n");
+
+	if (cold_rst_data.wd_rst_cause_arcpid)
+		preboot_puts("wd cause: ARCPID heartbeat\n");
+
+	if ((!cold_rst_data.wd_rst_cause_arcevent) && (!cold_rst_data.wd_rst_cause_arcpid))
+		preboot_puts("wd cause: ARCMGMT heartbeat\n");
+
+	cold_rst_data.wd_rst_cause_arcevent = 0;
+	cold_rst_data.wd_rst_cause_arcpid = 0;
+
+	regmap_writel(cold_rst_data.data, mmCOLD_RST_DATA);
+}
